core/src/main/java/org/elasticsearch/action/update/UpdateHelper.java

```diff
@@ -99,7 +100,7 @@ public Result prepare(UpdateRequest request, IndexShard indexShard) {
                     // we need to unwrap the ctx...
                     ctx = (Map<String, Object>) script.unwrap(ctx);
                 } catch (Exception e) {
-                    throw new IllegalArgumentException("failed to execute script", e);
+                    throw new IllegalArgumentException("Failed to execute script: " + ExceptionsHelper.detailedMessage(e), e);
```

Comment:
I think we should keep the message chain as it was for these exceptions? Someone wanting to find the root cause should iterate through the exception chain?